itest/lnd_open_channel_test.go

```diff
@@ -877,6 +877,115 @@ func testFundingExpiryBlocksOnPending(ht *lntest.HarnessTest) {
 	ht.MineBlocksAndAssertNumTxes(1, 1)
 }
 
+// testPendingChannelConfirmationUntilActive verifies the value for the rpc
+// field ConfirmationUntilActive updates correctly as soon as blocks are
+// confirmed.
+func testPendingChannelConfirmationUntilActive(ht *lntest.HarnessTest) {
+	var (
+		numConfs uint32         = 5
+		chanAmt  btcutil.Amount = 100000
+	)
+
+	// Since we want Bob's channels to require more than 1 on-chain
+	// confirmation before becoming active, we will launch Bob with the
+	// custom defaultchanconfs flag.
+	alice := ht.NewNodeWithCoins("Alice", nil)
+	bob := ht.NewNode("Bob", []string{
+		fmt.Sprintf("--bitcoin.defaultchanconfs=%v", numConfs),
+	})
+
+	// Ensure Alice and Bob are connected.
+	ht.EnsureConnected(alice, bob)
+
+	// Alice initiates a channel opening to Bob.
+	param := lntest.OpenChannelParams{Amt: chanAmt}
+	ht.OpenChannelAssertPending(alice, bob, param)
+
+	// Both Alice and Bob have one pending open channel.
+	ht.AssertNumPendingOpenChannels(alice, 1)
+	ht.AssertNumPendingOpenChannels(bob, 1)
+
+	// Since the funding transaction is not confirmed yet,
+	// ConfirmationsUntilActive will always be numConfs, and confirmation
+	// height will be 0.
+	require.Equal(ht, numConfs, ht.AssertNumPendingOpenChannels(alice,
+		1)[0].ConfirmationsUntilActive)
+	require.Equal(
+		ht, uint32(0),
+		ht.AssertNumPendingOpenChannels(alice, 1)[0].ConfirmationHeight,
+	)
+
+	require.Equal(ht, numConfs, ht.AssertNumPendingOpenChannels(bob, 1)[0].
+		ConfirmationsUntilActive)
+	require.Equal(
+		ht, uint32(0),
+		ht.AssertNumPendingOpenChannels(bob, 1)[0].ConfirmationHeight,
+	)
```

Comment:
It would be more readable this way, also later:
```golang
	// Both Alice and Bob have one pending open channel.
	alicePending := ht.AssertNumPendingOpenChannels(alice, 1)[0]
	bobPending := ht.AssertNumPendingOpenChannels(bob, 1)[0]

	// Since the funding transaction is not confirmed yet,
	// ConfirmationsUntilActive will always be numConfs, and confirmation
	// height will be 0.
	require.Equal(ht, numConfs, alicePending.ConfirmationsUntilActive)
	require.EqualValues(ht, 0, alicePending.ConfirmationHeight)

	require.Equal(ht, numConfs, bobPending.ConfirmationsUntilActive)
	require.EqualValues(ht, 0, bobPending.ConfirmationHeight)
```

All this may be flaky later, so maybe extract it into a helper function with a `wait.NoError` to wait for the expected values to appear?